### ISSUE TITLE
FP: foreach instead of for but with another condition

### ISSUE THREAD

**Issue by CZechBoY** [kalessil/phpinspectionsea#451]
Hi,
I get this warning: `Foreach can probably be used instead (easier to read and support; ensure a string is not iterated).`
for this code
```php
$arr = [];
$condition = 0;
for ($i = 0, $iMax = count($arr); $i < $iMax || $condition; $i++) {
    $condition = $arr[$i] % 10;
}
```

**Comment by funivan** [comment]
Thanks for the reports. It is definitely false positive.

**Comment by kalessil** [comment]
Thank you for reporting @CZechBoY, will be in tomorrow's release.
